### PR TITLE
Vam 414 vam 480 xus division get and set

### PR DESCRIPTION
**Pull request by kekuso**
no queries! just lots of context manipulation...

### CODE REVIEW

File: services/userMetaService/listDivisions.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    // if context.divisions has a single division that is the same as context.divisionId,
+    // then return success (make no changes)
+    if (currentDivisions.length === 1 && currentDivisions[0].id === divisionId) {
+        callback(null, { success: 'User & Context Division are the same, no changes.' });
+    } else if (currentDivisions.length === 1 && currentDivisions[0].id !== divisionId) {
+        // if context.divisions has a single division that is NOT the same as context.divisionId,
+        // then CHANGE to system division and return success
+        /* eslint-disable no-param-reassign */
+        context.divisions = {
```
Comment (mfuroyama) [inline services/userMetaService/listDivisions.js]: Correct me if I'm wrong, but I believe `context.divisions` should be an array...

File: services/userMetaService/listDivisions.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            stationNumber: context.systemDivisionStationNumber,
+            label: context.systemDivisionLabel,
+        };
+        context.divisionId = context.systemDivisionId;
+        context.divisionLabel = context.systemDivisionLabel;
+        /* eslint-enable no-param-reassign */
+        callback(null, { success: 'User division changed to System division.' });
+    } else if (currentDivisions.length > 1) {
+        // if user has multiple division properties, then return the list of divisions (sans the Mongo _id)
+        const divisionList = currentDivisions.filter((div) => {
```
Comment (mfuroyama) [inline services/userMetaService/listDivisions.js]: Here, consider using a combination of `map` and `_.omit`:
```Javascript
const divisionList = currentDivisions.map(div => _.omit(div, '_id'));
```

File: services/userMetaService/listDivisions.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        // if context.divisions has a single division that is NOT the same as context.divisionId,
+        // then CHANGE to system division and return success
+        /* eslint-disable no-param-reassign */
+        context.divisions = {
+            id: context.systemDivisionId,
+            stationNumber: context.systemDivisionStationNumber,
+            label: context.systemDivisionLabel,
+        };
+        context.divisionId = context.systemDivisionId;
+        context.divisionLabel = context.systemDivisionLabel;
```
Comment (mfuroyama) [inline services/userMetaService/listDivisions.js]: Consider using the lodash `_.extend` here to get rid of the ESLint warning:
```Javascript
_.extend(context, {
    divisions: [...],
    divisionId: context.systemDivisionId,
    divisionLabel: context.systemDivisionLabel,
});
```

File: services/userMetaService/setDivision.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    if (!context.divisions || !context.divisions.length) {
+        const divisionObj = {
+            id: context.systemDivisionId,
+            stationNumber: context.systemDivisionStationNumber,
+            label: context.systemDivisionLabel,
+        };
+        /* eslint-disable no-param-reassign */
+        context.divisions = [divisionObj];
+        context.divisionId = context.systemDivisionId;
+        context.divisionLabel = context.systemDivisionLabel;
```
Comment (mfuroyama) [inline services/userMetaService/setDivision.js]: Consider using the lodash `_.extend` here to get rid of the ESLint warning:
```Javascript
_.extend(context, {
    divisions: [...],
    divisionId: context.systemDivisionId,
    divisionLabel: context.systemDivisionLabel,
});
```
`_.extend` modifies the object in-place, so it should do what you need it to.

File: services/userMetaService/listDivisions.js (hunk 1 above)
Comment (kekuso) [reply services/userMetaService/listDivisions.js]: You right, you right 👍

**Pull request merged**